parsimony/functions/penalties.py

```diff
@@ -1235,10 +1244,10 @@ def L(self):
 
         lmaxA = self.lambda_max()
         # The (largest) Lipschitz constant of the gradient would be the operator
-        # norm of 2A'A, which thus is the square of the largest singular value
-        # of 2A'A.
+        # norm of 2A'A or 2L, which thus is the square of the largest singular value
+        # of 2A'A or largest eigenvalue of 2L.
 
-        return self.l * (2 * lmaxA) ** 2
+        return self.l * (2 * lmaxA) # I think the eigen value was squared twice
```

Comment:
Can you double check this?

Reply:
It depends if A (adjacency) or L (Laplacian) is given. As mentioned in the commented lines above, lmaxA should be squared if it is the singular value of the adjacency matrix, which is done below (l1269), but not squared if it is the eigenvalue of L, which is not clear to me, because a singular value of L is also computed (l1274).

Reply:
Note that lambda_max computes on A, and L computes on A'*A. Can you check (with small random matrices, for instance) that you got this right?

Reply:
done